File: server/src/main/java/com/skywalker/backend/service/repo/PatientService.java
package com.skywalker.backend.service.repo;

import com.skywalker.backend.model.Patient;
import com.skywalker.backend.dto.PatientDTO;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface PatientService {

    Patient getPatientById(Long id);

    List<Patient> getAllPatients();

    Patient createPatient(PatientDTO dto);

    Patient updatePatient(Long id, PatientDTO dto);

    void deletePatient(Long id);
}

File: client/src/services/patientService.js
import api from './api';

export const patientService = {
  getAllPatients: async () => {
    return await api.get('/api/patients');
  },

  getPatientById: async (id) => {
    return await api.get(`/api/patients/${id}`);
  },

  updatePatient: async (id, patientData) => {
    return await api.put(`/api/patients/${id}`, patientData);
  },

  deletePatient: async (id) => {
    return await api.delete(`/api/patients/${id}`);
  },
};
